### DISCUSSION TITLE
Ghostty increases stack size and file handle / descriptor open soft limits on Linux. Soft limits should be respected for child PTYs.

### DISCUSSION THREAD

**Discussion by tbiehn** [ghostty-org/ghostty#4215] [Feature Requests, Ideas] [answered]
On my machine Ghostty sets a soft stack size limit of 
`-s: stack size (kbytes)             16384` 
and a file handle limit of 
`-n: file descriptors                1073741816`. 
[src/file.zig](https://github.com/ghostty-org/ghostty/blob/47cf5cbb4023d70fc1009d1953fc9348fc1b60c8/src/os/file.zig#L7) mentions that file handle limits are maximized, ostensibly to allow a single parent process to own many children PTYs without hitting the handle limit. Cool.

Perhaps ghostty should not pass these soft limits to child PTYs.
Maybe we should save the unmodified limits and set them on child PTYs?

**Comment by tbiehn** [comment]
I don't know zig well enough to do a good job at this - but this shows how NOFILE and STACK can be restored in Command.zig after 'pre_exec' handlers execute in the POSIX subprocess.
[ghostty_NOFILE_STACK_patch.txt](https://github.com/user-attachments/files/18283600/ghostty_NOFILE_STACK_patch.txt)

**Comment by tbiehn** [comment]
I can't determine why STACK limits differ between Ghostty and Alacritty or TTY. Anyone know why it's roughly double?

**Comment by mitchellh** [comment, marked as answer]
Approved #4232

**Comment by tbiehn** [comment]
Opened PR https://github.com/ghostty-org/ghostty/pull/4242
